superset-frontend/plugins/plugin-chart-table/src/TableChart.tsx

```diff
@@ -629,7 +629,11 @@ export default function TableChart<D extends DataRecord = DataRecord>(
       const startPosition = value[0];
       const colSpan = value.length;
       // Retrieve the originalLabel from the first column in this group
-      const originalLabel = columnsMeta[value[0]]?.originalLabel || key;
+      const firstColumnInGroup = filteredColumnsMeta[value[0]];
```

Comment:
Can't we use startPosition here?
```suggestion
      const firstColumnInGroup = filteredColumnsMeta[startPosition];
```